controllers/namespace_mapping/namespace_mapping_controller.go

```diff
@@ -0,0 +1,240 @@
+	IMNodes       map[string]string
+}
+
+func applyForConfigMapName(ns string) predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(event event.CreateEvent) bool {
+			return event.Object.GetName() == namespaceMappingConfigMap && event.Object.GetNamespace() == ns
+		},
+		UpdateFunc: func(event event.UpdateEvent) bool {
+			return event.ObjectNew.GetName() == namespaceMappingConfigMap && event.ObjectNew.GetNamespace() == ns
+		},
+		DeleteFunc: func(event event.DeleteEvent) bool {
+			return event.Object.GetName() == namespaceMappingConfigMap && event.Object.GetNamespace() == ns
+		},
+	}
+}
+
+func Add(mgr manager.Manager, ns string) error {
+	return NewReconciler(mgr, ns).SetupWithManager(mgr, ns)
+}
+
+// NewReconciler returns a new ReconcileActiveGate
+func NewReconciler(mgr manager.Manager, ns string) *ReconcileNamespaceMapping {
+	return &ReconcileNamespaceMapping{
+		client:    mgr.GetClient(),
+		apiReader: mgr.GetAPIReader(),
+		namespace: ns,
+	}
+}
+
+func (r *ReconcileNamespaceMapping) SetupWithManager(mgr ctrl.Manager, ns string) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.ConfigMap{}).
+		WithEventFilter(applyForConfigMapName(ns)).
+		Complete(r)
+}
+
+func (r *ReconcileNamespaceMapping) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	r.logger.Info("Reconciling namespace map")
+
+	mappingConfigMap := &corev1.ConfigMap{}
+	err := r.client.Get(ctx, client.ObjectKey{Name: namespaceMappingConfigMap, Namespace: r.namespace}, mappingConfigMap)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+
+	codeModulesMapping := getCodeModulesNamespaceMapping(mappingConfigMap.Data)
+
+	kubeSystemUID, err := kubesystem.GetUID(r.apiReader)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	infraMonitoringNodes, err := r.getInfraMonitoringNodes()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err = r.replicateInitScriptAsSecret(codeModulesMapping, kubeSystemUID, infraMonitoringNodes); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileNamespaceMapping) replicateInitScriptAsSecret(namespaceMap []namespaceMapping, kubeSystemUID types.UID, infraMonitoringNodes map[string]string) error {
+	for _, mapping := range namespaceMap {
+		scriptData, err := r.prepareScriptForDynaKube(mapping.dynakube, kubeSystemUID, infraMonitoringNodes)
+		if err != nil {
+			return err
+		}
+
+		data, err := scriptData.generate()
+		if err != nil {
+			return err
+		}
+
+		if err = utils.CreateOrUpdateSecretIfNotExists(r.client, r.apiReader, webhook.SecretConfigName, mapping.namespace, data, corev1.SecretTypeOpaque, r.logger); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *ReconcileNamespaceMapping) prepareScriptForDynaKube(dk string, kubeSystemUID types.UID, infraMonitoringNodes map[string]string) (*script, error) {
```

Comment:
Nitpick: `dk` should be `dkName` since `dk` is usually used to describe the actual instance